core/src/Cuda/Kokkos_Cuda_ZeroMemset.hpp

```diff
@@ -0,0 +1,50 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+#ifndef KOKKOS_CUDA_ZEROMEMSET_HPP
+#define KOKKOS_CUDA_ZEROMEMSET_HPP
+
+#include <Kokkos_Macros.hpp>
+#ifdef KOKKOS_ENABLE_CUDA
```

Comment:
You don't need to guard for the Cuda backend being enabled.
The header needs to be self-contained though so you are missing header includes and/or forward definitions.

Reply:
"The header needs to be self-contained though"
IIRC, there is a test that should be checking all headers for being self-contained, I think it is : `core/unit_tests/headers_self_contained ` how come this was not caught?

Reply:
Because we only find the header files directly sitting in src/ not the ones in nested subdirectories
https://github.com/kokkos/kokkos/blob/aa1f48f3172069f212ed2c97b74b786999b26f17/core/unit_test/headers_self_contained/CMakeLists.txt#L6-L7

Reply:
Why is this the case? Would it make sense to add subdirectories?

Reply:
The headers in subdirectories are not necessarily self-contained (yet).

Reply:
You're welcome to give it a shot (in a separate pull request). 🙂

Reply:
Yeah, I'll give it a go.